### PR TITLE
(DOCSP-10236): Add Reference page for all available commands

### PR DESCRIPTION
**Pull request by jeff-allen-mongo**
Add Reference page for all available Commands for VSCE. 

A note for review, we're going to be tucking this page and the Settings page into a `Reference` ToC item once both individual PRs are approved.

Staged: https://docs-mongodbcom-staging.corp.mongodb.com/f57d85b/mongodb-vscode/docsworker/DOCSP-10236/commands

### CODE REVIEW

File: source/commands.txt [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+This document lists the VS Code commands available for
+|vsce|. To open the Command Palette and execute these commands:
+
+.. include:: /includes/list-tables/command-palette-options.rst
+
+Connection Commands
+-------------------
+
+Use these commands to manage connections to your MongoDB deployments.
```
Comment (rkumar-mongo) [inline source/commands.txt:23]: Should we specify that we intend users to issue these *from* the command palette explicitly? VSCode unfortunately uses the same rough UI for command palette vs search, and you can open up the latter just by missing the shift key.

Comment (rkumar-mongo) [inline source/commands.txt:23]: to create new connections *or* manage existing connections ?

File: source/commands.txt [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+   * - Command
+     - Description
+
+   * - ``Connect``
+     - Connect to your MongoDB deployment by filling in your connection
+       information in the connection dialog. To learn more about
+       connecting to a deployment, see :ref:`vsce-connect`.
+
+   * - ``Connect with Connection String``
+     - Connect to your MongoDB deployment by pasting your connection
```
Comment (rkumar-mongo) [inline source/commands.txt:38]: /s/connection string/MongoDB Connection String URI

or something so that we're specific as to what a connection string is. You could link to the manual page on the matter. 

We could also cheat and mention something like 'For example, MongoDB Atlas provides a connection string for your clusters you can pass into this command"

File: source/commands.txt [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+       string directly into the command palette. To learn more about
+       connecting to a deployment, see :ref:`vsce-connect`.
+
+   * - ``Disconnect``
+     - If you are currently connected to a MongoDB deployment,
```
Comment (rkumar-mongo) [inline source/commands.txt:43]: More of an implementation question, but we state 'deployments' up high, but 'deployment' here - do we support multiple connections? If so, how do I use this command to specify *one* deployment? Or is the meaning here that we disconnect from the *selected* deployment (i.e. i have to select a deployment, *then* run Disconnect)

File: source/commands.txt [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+.. note::
+
+   You can only have one active connection to a MongoDB
+   deployment open at a time.
+
+Shell Commands
+--------------
+
+Use this command to open a MongoDB Shell connected to your deployment.
```
Comment (rkumar-mongo) [inline source/commands.txt:58]: This presumes that MongoDB Shell is installed on the system, doesn't it? We might link to the installation docs on the matter. Though - do we link to the shell, mcli, mongosh? If we are bundling the shell in with the extension, I wonder if we should mention somewhere what the expected shell version is (or something)

File: source/commands.txt [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+       By default, playgrounds load with a template containing
+       sample commands. You can disable this template from your
+       :ref:`extension settings <vsce-settings>`.
+
+   * - ``Run All From Playground``
+     - Run file you are currently viewing in VS Code as a playground.
+       |vsce| outputs the results of your playground in the
+       :guilabel:`Output` view in VS Code.
+
+MongoDB View Commands
```
Comment (rkumar-mongo) [inline source/commands.txt:96]: So I could see a user getting confused and thinking this has something to do with Views as a container structure, rather than a |vsce| view. Though that is mitigated by stating it's a |vsce| view immediately after the jump, I suppose?

File: source/commands.txt [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+|vsce|. To open the Command Palette and execute these commands:
+
+.. include:: /includes/list-tables/command-palette-options.rst
+
+Connection Commands
+-------------------
+
+Use these commands to manage connections to your MongoDB deployments.
+
+.. list-table::
```
Comment (rkumar-mongo) [inline source/commands.txt:25]: So, a general thought - I have (personally) found that tables like this can be a little hard to parse, especially if they get longer. Also, we can't link to any of them.

I wonder if it would be worth creating a directive for `vsce-command` or something similar so that we can link to these easily.

Alternatively, you can cheat and force these into anchors by doing:

```
- .. _vsce-command-connect:
  :ref:`Connect <vsce-command-connect>`
- Connect to your...

```

You lose the monospacing, but that should be OK (I think you can use xmlrole if you *really* want monospacing).

**Comment by jeff-allen-mongo** [comment]
Thanks Ravind. Updated staging is here: https://docs-mongodbcom-staging.corp.mongodb.com/45b9161/mongodb-vscode/docsworker/DOCSP-10236/commands

File: source/commands.txt (hunk 3 above)
Comment (rkumar-mongo) [reply source/commands.txt:61]: I think we need to clarify between creating connections and having an 'Active' connection. I'm assuming 'Active' connection is the one selected in the MongoDB View inside VSCode?

i.e. creating a connection adds a new entry to the MongoDB View. 
Selecting a connection activates it.
The Disconnect/Remove command removes the active/selected connection.

File: source/commands.txt [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+       disconnect from that deployment.
+
+   * - :ref:`Remove Connection <vsce-cmd-remove-connection>`
+
+       .. _vsce-cmd-remove-connection:
+
+     - Select and remove a connection from your list of connections
+       in |vsce|.
+
+.. note::
```
Comment (rkumar-mongo) [inline source/commands.txt:71]: I think we should lead with this, and clarify - you can have many *configured* connections, but only one *active* connection at a time.

File: source/commands.txt [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+       .. _vsce-cmd-launch-shell:
+
+     - .. important::
+
+          This command is not supported on Windows.
+     
+       Open the VS Code :guilabel:`Terminal` and automatically connect a
+       MongoDB shell to the deployment where |vsce| is connected. You
+       can set which shell this command uses in your ``mdb.shell``
+       :ref:`setting <vsce-settings>`.
```
Comment (rkumar-mongo) [inline source/commands.txt:100]: Maybe merge the second sentence with 102, like:

"Specify the MongoDB shell to use in the ``mdb.shell`` setting. The MongoDB |vsce| extension supports the following shells:"

And then maybe include an example string for how to specify that shell (is it just `mongo` or `mongosh` ? Like, the value should be the exact string you would pass into the terminal?

File: source/commands.txt [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+
+   * - Command
+     - Description
+
+   * - :ref:`Focus on MongoDB View <vsce-cmd-focus>`
+
+       .. _vsce-cmd-focus:
+
+     - Open the |vsce| view and change the focus of VS Code to
+       the extension.
```
Comment (rkumar-mongo) [inline source/commands.txt:157]: Maybe "Open and focus on the |vsce| view?

File: source/commands.txt (hunk 3 above)
Comment (jeff-allen-mongo) [reply source/commands.txt:61]: All good points. Yes, 'Active' is referring to the one which is currently selected. I added some more detail to the introduction of this section.

File: source/commands.txt (hunk 7 above)
Comment (jeff-allen-mongo) [reply source/commands.txt:71]: Moved it to the lead-in.

File: source/commands.txt (hunk 8 above)
Comment (jeff-allen-mongo) [reply source/commands.txt:100]: Agreed on the merge. The setting is a dropdown, so I think no clarification necessary.

File: source/commands.txt (hunk 9 above)
Comment (jeff-allen-mongo) [reply source/commands.txt:157]: I wasn't sure if "focus" would be commonly understood when phrased as such, but your suggestion is definitely cleaner so I'll go with it.

**Comment by jeff-allen-mongo** [comment]
Updated staging: https://docs-mongodbcom-staging.corp.mongodb.com/f57d85b/mongodb-vscode/docsworker/DOCSP-10236/commands

File: source/commands.txt [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+This document lists the VS Code commands available for
+|vsce|. To open the Command Palette and execute these commands:
+
+.. include:: /includes/list-tables/command-palette-options.rst
+
+.. include:: /includes/admonitions/tip-mongodb-command-palette.rst
```
Comment (rkumar-mongo) [inline source/commands.txt:20]: Nit: suggest including that 'in development' admonition from the install page here.

Comment (jeff-allen-mongo) [reply source/commands.txt:20]: Oh oops thought I had it. Thanks for flagging

**Comment by jeff-allen-mongo** [comment]
@mmarcon, this Command reference is ready for review. Feel free to punt to someone else if you like 😄
